Error: Issue with `EffectiveGasPrice` in `finalizer.go`

## Description:
While working with the zkevm-node/sequencer, the following error occurs:

``` plaintext
2023-08-23T10:50:45.057Z        WARN    sequencer/finalizer.go:573      EffectiveGasPrice is disabled, but failed to calculate BreakEvenGasPrice: L1 gas price 0        {"pid": 1, "version": "v0.2.5-30-g06ec9bf9", "txHash": "0x37765f6756816cf7b9f1cd9c607a52dd2936c21413ad850331bf1db918b2171a", "batchNumber": 1}
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).processTransaction
        /src/sequencer/finalizer.go:573
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).finalizeBatches
        /src/sequencer/finalizer.go:338
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).Start
        /src/sequencer/finalizer.go:182
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1233c47]

goroutine 140 [running]:
math/big.(*Int).Uint64(...)
        /usr/local/go/src/math/big/int.go:379
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).processTransaction(0xc0001ab800, {0x19d5aa8, 0xc000042040}, 0xc0007c6000)
        /src/sequencer/finalizer.go:580 +0x627
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).finalizeBatches(0xc0001ab800, {0x19d5aa8, 0xc000042040})
        /src/sequencer/finalizer.go:338 +0x59f
github.com/0xPolygonHermez/zkevm-node/sequencer.(*finalizer).Start(0xc0001ab800, {0x19d5aa8?, 0xc000042040}, 0xc000188300, 0xc0007e8300)
        /src/sequencer/finalizer.go:182 +0x2f9
created by github.com/0xPolygonHermez/zkevm-node/sequencer.(*Sequencer).Start
        /src/sequencer/sequencer.go:113 +0x72c
```

## Potential Solution:
It appears that the error might be related to a missing check. On line `580` of `finalizer.go`, the check should be modified as:

```go
if tx.BreakEvenGasPrice != nil && tx.BreakEvenGasPrice.Uint64() != 0 {
```

Instead of the current check:

```go
if tx.BreakEvenGasPrice.Uint64() != 0 {
```

--- a/sequencer/finalizer.go
+++ b/sequencer/finalizer.go
@@ -578,7 +578,7 @@ func (f *finalizer) processTransaction(ctx context.Context, tx *TxTracker) (errW
 
 		effectivePercentage := state.MaxEffectivePercentage
 
-		if tx.BreakEvenGasPrice.Uint64() != 0 {
+		if tx.BreakEvenGasPrice != nil && tx.BreakEvenGasPrice.Uint64() != 0 {
 			// If the tx gas price is lower than the break even gas price, we process the tx with the user gas price (100%)
 			if tx.GasPrice.Cmp(tx.BreakEvenGasPrice) <= 0 {
 				tx.IsEffectiveGasPriceFinalExecution = true

